test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -449,15 +500,536 @@ def test_last_used_time(self):
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[2])
+        self.server.expect(JOB, {'job_state': 'Q'}, id=j4id)
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[3])
+        self.cleanup_power_on()
+
+    @timeout(1500)
+    def test_preemption_sleep_nodes(self):
+        """
+        Test high priority job preempts a running job rather
+        than waiting for a node to be brought up.
+        """
+        self.scheduler.set_sched_config({'strict_ordering': 'True ALL'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '4'})
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.manager(MGR_CMD_SET, NODE,
+                                    {"poweroff_eligible": True}, n['id'])
+        a = {"power_on_off_enable": True,
+             "max_concurrent_nodes": "30", 'node_idle_limit': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 30}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        t = int(time.time())
+        self.logger.info("Waiting for 4 mins to poweroff all the nodes")
+        time.sleep(240)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_off", starttime=t)
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.expect(NODE, {'state': 'sleep'}, id=n['id'])
+        a = {"node_idle_limit": "1800", 'min_node_down_delay': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+
+        jid1 = self.submit_job(1000, {'Resource_List.vnode': self.names[0],
+                                      'Resource_List.vntype': 'cray_compute'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        t = int(time.time())
+        self.logger.info("Waiting for 10 mins to poweron the calandered nodes")
+        time.sleep(600)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_on", starttime=t)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(NODE, {'state': 'job-exclusive'}, id=self.names[0])
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True',
+             'priority': '150'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq1')
+        t = int(time.time())
+        jid2 = self.submit_job(1000, {ATTR_queue: 'wq1',
+                                      'Resource_List.vntype': 'cray_compute'})
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid1)
+        self.scheduler.log_match("Job preempted by requeuing", starttime=t)
```

Comment:
You didn't change the sched_config parameter sched_order.  It's SCR by default.  This means you'll suspend if you can first.  If suspension is enabled on your cray, your test will fail.  You want to set preempt_order to 'R'